File: src/nuzzgraph/server/core/JettyServer.java
package nuzzgraph.server.core;

import org.eclipse.jetty.server.Server;
import org.eclipse.jetty.server.handler.ContextHandlerCollection;
import org.eclipse.jetty.server.Handler;

public class JettyServer 
{

	private Server server;
	private ServerHandler handler;
        
	public JettyServer() 
	{
            this(604);
	}
	public JettyServer(Integer runningPort) 
	{
            server = new Server(runningPort);
            handler = new ServerHandler();
            server.setHandler(handler);
	}
	
	public void setHandler(ContextHandlerCollection contexts) 
	{
            server.setHandler(contexts);
	}
	
	public void start() throws Exception
	{
            server.start();
	}
	
	public void stop() throws Exception 
	{
            server.stop();
            server.join();
	}
	
	public boolean isStarted() 
	{
            return server.isStarted();
	}
	
	public boolean isStopped() 
	{
            return server.isStopped();
	}
}

File: src/nuzzgraph/server/core/JettyServer.scala
package nuzzgraph.server.core

import org.eclipse.jetty.server.Server
import org.eclipse.jetty.server.handler.ContextHandlerCollection

class JettyServer(runningPort:Int) {
  def this() = this(604) //default port

  val server : Server = new Server(runningPort)
  val handler : ServerHandler = new ServerHandler
  server.setHandler(handler)

  def setHandler(contexts: ContextHandlerCollection): Unit = {
    server.setHandler(contexts)
  }

  def start: Unit = {
    server.start
  }

  def stop: Unit = {
    server.stop
    server.join
  }

  def isStarted: Boolean = {
    return server.isStarted
  }

  def isStopped: Boolean = {
    return server.isStopped
  }
}
